test/e2e/kubernetes/kubernetes_test.go

```diff
@@ -125,23 +125,20 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
 		})
 
 		It("should have tiller running", func() {
-			if eng.HasTiller() {
+			if hasTiller, tillerAddon := eng.HasTiller(); hasTiller {
 				running, err := pod.WaitOnReady("tiller", "kube-system", 3, 30*time.Second, cfg.Timeout)
 				Expect(err).NotTo(HaveOccurred())
 				Expect(running).To(Equal(true))
-			} else {
-				Skip("tiller disabled for this cluster, will not test")
-			}
-		})
-
-		It("should have a tiller max-history of 5", func() {
-			if eng.HasTiller() {
-				pods, err := pod.GetAllByPrefix("tiller-deploy", "kube-system")
-				Expect(err).NotTo(HaveOccurred())
-				// There is only one tiller pod and one container in that pod.
-				actualTillerMaxHistory, err := pods[0].Spec.Containers[0].GetEnvironmentVariable("TILLER_HISTORY_MAX")
-				Expect(err).NotTo(HaveOccurred())
-				Expect(actualTillerMaxHistory).To(Equal("5"))
+				if tillerAddon.Config != nil {
```

Comment:
Would this ever be nil? What happens if it is?

Reply:
The current configuration enforces always non-nil:

https://github.com/Azure/acs-engine/blob/master/pkg/acsengine/defaults.go#L752

So the above is just for good hygiene.